src/compiler/scala/tools/nsc/Driver.scala

```diff
@@ -9,16 +9,19 @@ abstract class Driver {
 
   val prompt = residentPromptString
 
-  var reporter: ConsoleReporter = _
+  var reporter: Reporter = _
   protected var command: CompilerCommand = _
   protected var settings: Settings = _
 
+  /** Forward errors to the (current) reporter. */
   protected def scalacError(msg: String): Unit = {
     reporter.error(FakePos("scalac"), msg + "\n  scalac -help  gives more information")
   }
 
+  /** True to continue compilation. */
   protected def processSettingsHook(): Boolean = {
-    if (settings.version) { reporter echo versionMsg ; false } else true
+    if (settings.version) { reporter echo versionMsg ; false }
+    else !reporter.hasErrors
```

Comment:
It seems that `reporter.hasErrors` now implies `reporter.flush()` won't get called, whereas it used to.

See snippet below:
```
  if (processSettingsHook()) {
       val compiler = newCompiler()
+      reporter     = compiler.reporter    // adopt the configured reporter
       try {
         if (reporter.hasErrors)
           reporter.flush()
```